File: src/main/java/org/fx/debugger/context/UIContext.java
package org.fx.debugger.context;

import java.nio.charset.StandardCharsets;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import lombok.SneakyThrows;
import lombok.val;
import org.fx.debugger.constant.FXMLView;
import org.fx.debugger.view.ScenicView;

public enum UIContext {
  INSTANCE;

  @SneakyThrows
  public Parent load(FXMLView view) {
    val url = view.getView() + ".fxml";
    val loader = new FXMLLoader(ScenicView.class.getResource(url));
    loader.setControllerFactory(DIContext.INSTANCE::get);
    loader.setCharset(StandardCharsets.UTF_8);
    return loader.load();
  }
}

File: src/main/kotlin/org/scenicview/UIContext.kt
package org.scenicview

import io.github.oshai.kotlinlogging.KotlinLogging
import java.nio.charset.StandardCharsets
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import org.scenicview.constant.FXMLView

object UIContext {

  private val log = KotlinLogging.logger {}

  fun load(fxmlKey: FXMLView): Parent {
    val url = "${fxmlKey.view}.fxml"
    val loader = FXMLLoader(ScenicView::class.java.getResource(url))
    loader.setControllerFactory { DIContext.get(it) }
    loader.charset = StandardCharsets.UTF_8
    return loader.load()
  }
}
